zk_toolbox/crates/zk_inception/src/commands/chain/init.rs

```diff
@@ -69,6 +79,26 @@ pub async fn init(
 
     let mut general_config = chain_config.get_general_config()?;
     apply_port_offset(init_args.port_offset, &mut general_config)?;
+    let ports = ports_config(&general_config).context(MSG_PORTS_CONFIG_ERR)?;
+
+    let consensus_keys = generate_consensus_keys();
+    let genesis_spec = Some(get_genesis_specs(chain_config, &consensus_keys));
+
+    let public_addr = format!("{}:{}", CONSENSUS_PUBLIC_ADDRESS_HOST, ports.consensus_port);
```

Comment:
instead of going through string format, just use the constructors directly: https://doc.rust-lang.org/stable/std/net/enum.SocketAddr.html#method.new